aspnetcore/fundamentals/hosting.md

```diff
@@ -1,225 +1,534 @@
+You can set the *Environment* to any value. Framework-defined values include `Development`, `Staging`, and `Production`. Values aren't case sensitive. By default, the *Environment* is read from the `ASPNETCORE_ENVIRONMENT` environment variable. When using [Visual Studio](https://www.visualstudio.com/), environment variables may be set in the *launchSettings.json* file. For more information, see [Working with Multiple Environments](xref:fundamentals/environments).
 
-Note: When your app runs with Kestrel and IIS, the default behavior is to capture startup errors. 
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 ```csharp
-new WebHostBuilder()
-    .CaptureStartupErrors(true)
-   ```
+WebHost.CreateDefaultBuilder(args)
+    .UseEnvironment("Development")
+    ...
+```
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+```csharp
+var host = new WebHostBuilder()
+    .UseEnvironment("Development")
+    ...
+```
+
+---
 
-**Content Root** `string`
+### Hosting Startup Assemblies
 
-Key: `contentRoot`. Defaults to the folder where the application assembly resides (for Kestrel; IIS will use the web project root by default). This setting determines where ASP.NET Core will begin searching for content files, such as MVC Views. Also used as the base path for the [Web Root Setting](#web-root-setting). Set using the `UseContentRoot` method. Path must exist, or host will fail to start.
+Sets the app's hosting startup assemblies.
+
+**Key**: hostingStartupAssemblies  
+**Type**: *string*  
+**Default**: Empty string  
+**Set using**: `UseSetting`
+
+A semicolon-delimited string of hosting startup assemblies to load on startup. This feature is new in ASP.NET Core 2.0.
+
+Although the configuration value defaults to an empty string, the hosting startup assemblies always include the app's assembly. When you provide hosting startup assemblies, they're added to the app's assembly for loading when the app builds its common services during startup.
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 ```csharp
-new WebHostBuilder()
-    .UseContentRoot("c:\\mywebsite")
-   ```
+WebHost.CreateDefaultBuilder(args)
+    .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, "assembly1;assembly2")
+    ...
+```
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+```csharp
+var host = new WebHostBuilder()
+    .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, "assembly1;assembly2")
+    ...
+```
+
+---
 
-**Detailed Errors** `bool`
+### Prefer Hosting URLs
 
-Key: `detailedErrors`. Defaults to `false`. When `true` (or when Environment is set to "Development"), the app will display details of startup exceptions, instead of just a generic error page. Set using `UseSetting`.
+Indicates whether the host should listen on the URLs configured with the `WebHostBuilder` instead of those configured with the `IServer` implementation.
+
+**Key**: preferHostingUrls  
+**Type**: *bool* (`true` or `1`)  
+**Default**: true  
+**Set using**: `UseSetting`
```

Comment:
Can be set via `PreferHostingUrls` instead of only `UseSetting`
https://github.com/aspnet/Hosting/blob/rel/2.0.0/src/Microsoft.AspNetCore.Hosting.Abstractions/HostingAbstractionsWebHostBuilderExtensions.cs#L157. Though it uses `UseSetting` under the hood anyway.